### ISSUE TITLE
systemd service vector fails to start, PreStart validate does not use /etc/default/vector

### ISSUE THREAD

**Issue by pgassmann** [timberio/vector#8560]
### Vector Version

```
vector --version
vector 0.15.1 (x86_64-unknown-linux-gnu c5e4732 2021-07-27)
```

### Vector Configuration File

`/etc/vector/vector.toml `
```toml
# Set global options
data_dir = "/var/lib/vector"

```
`/etc/default/vector`
```
# cat /etc/default/vector
# /etc/default/vector
# This file can theoretically contain a bunch of environment variables
# for Vector.  See https://vector.dev/docs/setup/configuration/#environment-variables
# for details.
VECTOR_CONFIG_TOML=/etc/vector/*.toml
```

actual config (source and sink) is in `/etc/vector/docker-to-loki.toml`

### Expected Behavior

`systemctl start vector` should work

### Actual Behavior

`systemctl start vector` fails because the validate `ExecStartPre=/usr/bin/vector validate` command does not check the full config

### Additional Context

```
# systemctl status vector
● vector.service - Vector
   Loaded: loaded (/lib/systemd/system/vector.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2021-08-02 19:17:44 CEST; 6s ago
     Docs: https://vector.dev
  Process: 11684 ExecStartPre=/usr/bin/vector validate (code=exited, status=78)

Aug 02 19:17:44 youtrack1 systemd[1]: Starting Vector...
Aug 02 19:17:44 youtrack1 vector[11684]: Failed to load ["/etc/vector/vector.toml"]
Aug 02 19:17:44 youtrack1 vector[11684]: ------------------------------------------
Aug 02 19:17:44 youtrack1 vector[11684]: x No sources defined in the config.
Aug 02 19:17:44 youtrack1 vector[11684]: x No sinks defined in the config.
Aug 02 19:17:44 youtrack1 systemd[1]: vector.service: Control process exited, code=exited status=78
Aug 02 19:17:44 youtrack1 systemd[1]: vector.service: Failed with result 'exit-code'.
Aug 02 19:17:44 youtrack1 systemd[1]: Failed to start Vector.
```

### References

This is a regression introduced by

- #7256

this change made vector unable to start. the validate command does not consider the environment in `/etc/default/vector`
I had no config in `/etc/vector/vector.toml`, validate would report that there are no sinks and no sources defined.

**Comment by pgassmann** [comment]
in short:

`vector validate` should respect the same environment variables for loading config files as `vector`

**Comment by pgassmann** [comment]
My Workaround: disable validates command in override:

`/etc/systemd/system/vector.service.d/override.conf`

```toml
[Service]
#ExecStartPre=/usr/bin/vector validate
ExecStartPre=
# ExecStart=/usr/bin/vector
# ExecReload=/usr/bin/vector validate
ExecReload=
ExecReload=/bin/kill -HUP $MAINPID
```

run `systemctl daemon-reload` after changing that file. or use `systemctl edit vector.service` and paste the content

**Comment by nnz1024** [comment]
Looks like problem is in `vector validate` command (it does not respect the environment), not in systemd nor service file.

Just tested with the simple unit
```
test:~ # cat /etc/systemd/system/test.service 
[Service]
EnvironmentFile=-/etc/default/vector
ExecStartPre=/usr/bin/env
ExecStart=/bin/true
Type=simple
test:~ # cat /etc/default/vector             
# /etc/default/vector
# This file can theoretically contain a bunch of environment variables
# for Vector.  See https://vector.dev/docs/setup/configuration/#environment-variables
# for details.
TEST123=456
test:~ # systemctl daemon-reload && systemctl start test.service && systemctl status test.service   
● test.service
   Loaded: loaded (/etc/systemd/system/test.service; static; vendor preset: enabled)
   Active: inactive (dead)

авг 03 15:23:40 dtest-x1 systemd[1]: Starting test.service...
авг 03 15:23:40 dtest-x1 env[7928]: LANG=ru_RU.UTF-8
авг 03 15:23:40 dtest-x1 env[7928]: PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
авг 03 15:23:40 dtest-x1 env[7928]: INVOCATION_ID=60613516f3f3436cb5ebeaddf6dc9600
авг 03 15:23:40 dtest-x1 env[7928]: JOURNAL_STREAM=9:17718856
авг 03 15:23:40 dtest-x1 env[7928]: TEST123=456
авг 03 15:23:40 dtest-x1 systemd[1]: Started test.service.
авг 03 15:23:40 dtest-x1 systemd[1]: test.service: Succeeded.
```

So, command in ExecStartPre receives variables from EnvironmentFile.

**Comment by spencergilbert** [comment]
Looking into a fix for this now!

**Comment by spencergilbert** [comment]
A fix has been merged, and we'll be cutting a 0.15.2 release on the 10th which will include the fix for the `validate` command
